src/pymovements/stimulus/text.py

```diff
@@ -80,6 +80,24 @@ def __init__(
         self.end_y_column = end_y_column
         self.page_column = page_column
 
+    def split(
+            self,
+            by: str,
+    ) -> list[TextStimulus]:
+        """Split the AOI df.
+
+        Parameters
+        ----------
+        by: str
+            Splitting criteria.
+
+        Returns
+        -------
+        list[TextStimulus]
+            A list of TextStimulus objects.
+        """
+        return self.aois.partition_by(by=by, as_dict=False)
```

Comment:
I must have missed this in my review two weeks ago.

this line returns a list of `polars.DataFrame` not a list of `TextStimulus` objects.

You should probably do something like this:

```python
return [
    TextStimulus(
        aois=df,
        aoi_column=self.aoi_column,
        width_column=self.width_column,
        ...,  # remaining fields
        page_column=self.page_column,
    )
    for df in self.aois.partition_by(by=by, as_dict=False)
]
```

Reply:
@SiQube I am confused why mypy didn't complain here, as I would expect a type mismatch. I am clueless. Do you have an explanation? I am a bit worried that there's something wrong in our CI.

Reply:
I actually don't know, I'll try to figure out.